packages/datagrid/src/components/RichCellEditor/RichCellEditor.component.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useRef, useState } from 'react';
+import { SkeletonParagraph } from '@talend/design-system';
+import classNames from 'classnames';
+
+import { AgGridCellValue } from '../../types';
+
+import CellEditorDatalist from './components/CellEditorDatalist.component';
+import CellEditorTextarea from './components/CellEditorTextarea.component';
+
+import theme from './RichCellEditor.component.scss';
+
+type CellValue = string;
+
+interface RichCellEditorPropTypes {
+	initialValue: CellValue;
+	isLoading?: boolean;
+	hasSuggestions?: boolean;
+	eGridCell: HTMLDivElement;
+	onChange: (value: CellValue) => void;
+	onFilter?: (search: string) => Promise<AgGridCellValue[]>;
+	onCancel: () => void;
+}
+
+function RichCellEditor(props: RichCellEditorPropTypes) {
+	const { onChange, onFilter, onCancel, initialValue, hasSuggestions, isLoading, eGridCell } =
+		props;
+	const [value, setValue] = useState(initialValue);
+
+	const previousCellColor = useRef<string>();
+	React.useEffect(() => {
+		previousCellColor.current = eGridCell.style.color;
+		eGridCell.style.color = 'transparent';
+		return () => {
+			eGridCell.style.color = previousCellColor.current!;
+		};
+	}, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Ugly solution. This is used to hide the cell content when the cell editor is opened but when the content editor is not yet shown (while loading semantic types for example).